### PR TITLE
RestoreDashboards: Add link to dashboards browse view

### LINKED ISSUE #93885: RestoreDashboards: Add link to dashboards browse view
As the Recently deleted page is hard to find we need to link the page.
Consult with @teddyba whether the suggested place is acceptable.

**Requirements:**
- secondary button
- button shown under the same conditions as Recently deleted page
- add `reportInteraction()`
- in the browse view for dashboards and the folder view (see below)

Image
Image

### PR DESCRIPTION
**Pull request by L-M-K-B**
**What is this feature?**
I'm adding a button to the browse dashboards page in order to give a hint where to find the new section.
Screenshot 2024-09-30 at 14 32 35
Screenshot 2024-09-30 at 14 32 51

**Which issue(s) does this PR fix?**:
Fixes #93885 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Comment by teddyba** [comment]
Let's change the copy to "Recently deleted" instead.

**Comment by teddyba** [comment]
I also commented in https://github.com/grafana/grafana/issues/93885. 

I have another thought: 
Could we add "View recently deleted dashboards" as an action under the "Folder actions" button instead -- assuming that this action is not super common and therefore doesn't deserve so much real estate?

**Comment by L-M-K-B** [comment]
@teddyba 

> Let's change the copy to "Recently deleted" instead.

Screenshot 2024-10-01 at 17 29 46
Screenshot 2024-10-01 at 17 30 06


> Could we add "View recently deleted dashboards" as an action under the "Folder actions" button instead

Screenshot 2024-10-01 at 17 44 15

**Comment by teddyba** [comment]
@eledobleefe Thank you for mocking these up -- super helpful!! 

I'm going to lay out the pros and cons of each suggested approach to aid us in a decision.

1. **Option 1: Introduce "Recently deleted" buttons on both the Dashboards level and the Folder level.** 
    - Pros: Discoverable in both levels where a user might be looking for a recently deleted dashboard. Also, implementing the button will help us understand how commonly users are visiting the deleted dashboards.
    - Cons: It is strange that we use a button for a link, when the other buttons are actions. Also, I still question how much discoverability "deleted dashboards" deserves.

2. **Option 2: Add "View recently deleted dashboards" as an option under "Folder Actions"** 
     - Pros: Less obtrusive than Option 1 above. 
    - Cons: Only discoverable from the Folder level, and I don't know where users will most commonly be wanting to navigate to deleted dashboards from -- from the Dashboards page or from within a folder, when they are looking for a dashboard and don't find it? 

3. **Option 3: Create a "fake" folder that links to Recently deleted dashboards.**
     - Pros: Less obtrusive than Option 1. 
    - Cons: Only discoverable from the Dashboards level (the reverse of the con for Option 2 above).  

4. **Option 4: Do nothing.**
     - Pros: We avoid potential consequences of taking up too much real estate with the other proposed solutions? Also, less effort and can ship sooner. 
    - Cons: We won't learn anything about the level of discoverability needed, or from where it makes most sense to link the user to recently deleted.

**Comment by teddyba** [comment]
Based on the options laid out above, I definitely don't want to go with Option 4, because we won't learn anything. I am now leaning towards option 1, to have the button in both places, because then we can track and learn: 
- How often folks are clicking the button
- From which level (Dashboards or a single folder level) users want to go see recently deleted dashboards

**Comment by L-M-K-B** [comment]
Hi @teddyba, thanks a lot for the explanations. 
I'll keep the button and improve the tracking because at the moment it's not distinguishing between dashboards and folder view.

File: public/app/features/browse-dashboards/BrowseDashboardsPage.tsx [hunk 1]
```diff
@@ -81,6 +83,7 @@ const BrowseDashboardsPage = memo(() => {
   const { data: rootFolder } = useGetFolderQuery('general');
   let folder = folderDTO ? folderDTO : rootFolder;
   const { canEditFolders, canEditDashboards, canCreateDashboards, canCreateFolders } = getFolderPermissions(folder);
+  const hasAdminRights = contextSrv.user.orgRole === OrgRole.Admin;
```
Comment (Clarity-89) [inline public/app/features/browse-dashboards/BrowseDashboardsPage.tsx:86]: Should we check for Grafana admin as well? A user can technically be a Grafana admin and not an org admin.

Comment (Clarity-89) [reply public/app/features/browse-dashboards/BrowseDashboardsPage.tsx:86]: Smth like [this](https://github.com/grafana/grafana/blob/dd7f45011da05f58ccb51d67049199c3b6f7ce9a/public/app/features/alerting/unified/utils/misc.ts#L216), for example.

**Review by ashharrison90 — COMMENTED** [review]
question about subpaths

File: public/app/features/browse-dashboards/BrowseDashboardsPage.tsx (hunk 1 above)
Comment (L-M-K-B) [reply public/app/features/browse-dashboards/BrowseDashboardsPage.tsx:86]: @Clarity-89 In the BE the feature is set up to be available like this: `c.SignedInUser.GetOrgRole() == org.RoleAdmin` (see [here](https://github.com/grafana/grafana/blob/9144e3b44a2327b9ce30c75c2ec258264f510585/pkg/services/navtree/navtreeimpl/navtree.go#L390)). If we want the feature to be available for Grafana admins I assume we need to change the condition in the BE as well. 
What's the actual difference between Grafana admin and org admin?

File: public/app/features/browse-dashboards/BrowseDashboardsPage.tsx [hunk 2]
```diff
@@ -104,13 +107,28 @@ const BrowseDashboardsPage = memo(() => {
...
     <Page
       navId="dashboards/browse"
       pageNav={navModel}
       onEditTitle={showEditTitle ? onEditTitle : undefined}
       actions={
         <>
+          {config.featureToggles.dashboardRestore && config.featureToggles.dashboardRestoreUI && hasAdminRights && (
+            <LinkButton
+              variant="secondary"
+              href={'/dashboard/recently-deleted'}
```
[the remark this answers is not in the document]
Comment (L-M-K-B) [reply public/app/features/browse-dashboards/BrowseDashboardsPage.tsx:126]: I don't know but to make sure it works I'll add this `getConfig().appSubUrl`.

File: public/app/features/browse-dashboards/BrowseDashboardsPage.tsx (hunk 1 above)
Comment (Clarity-89) [reply public/app/features/browse-dashboards/BrowseDashboardsPage.tsx:86]: Looks like I've missed that one myself 😅  The condition should be `if s.features.IsEnabled(c.Req.Context(), featuremgmt.FlagDashboardRestoreUI) && (c.SignedInUser.GetOrgRole() == org.RoleAdmin || c.IsGrafanaAdmin) {`. 

Grafana Admin is an administrator for the whole server, I think this is more relevant in the on-prem case. There's some info about this in the [docs](https://grafana.com/docs/grafana/latest/administration/user-management/server-user-management/assign-remove-server-admin-privileges/).

File: public/app/features/browse-dashboards/BrowseDashboardsPage.tsx [hunk 3]
```diff
@@ -104,13 +107,28 @@ const BrowseDashboardsPage = memo(() => {
     }
   };
 
+  const handleButtonClickToRecentlyDeleted = () => {
+    reportInteraction('grafana_browse_dashboards_page_button_to_recently_deleted', {
+      origin: window.location.pathname === '/dashboards' ? 'Dashboards' : 'Folder view',
```
[the remark this answers is not in the document]
Comment (L-M-K-B) [reply public/app/features/browse-dashboards/BrowseDashboardsPage.tsx:112]: I don't know but to make sure it works I'll add this getConfig().appSubUrl.

**Review by Clarity-89 — APPROVED** [review]

**Pull request merged**